extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -368,6 +368,10 @@ export abstract class AzureAuth implements vscode.Disposable {
 		try {
 			Logger.verbose('Fetching tenants', tenantUri);
 			const tenantResponse = await this.makeGetRequest(tenantUri, token.token);
+			if (tenantResponse.status !== 200) {
+				Logger.error(`Error with tenant response, status: ${tenantResponse.status} | status text: ${tenantResponse.statusText}`);
```

Comment:
We can avoid further activity in this method once failed response is identified, so you can simply throw an exception from here with this message. Only if status == 200, we should be processing any further.